styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -62,4 +72,46 @@ public FlyteExecution createExecution(final String name, final FlyteExecConf fly
       throw new CreateExecutionException(flyteExecConf, e);
     }
   }
+
+  public void poll(final String project, final String domain, final String name, RunState runState) {
+    Objects.requireNonNull(project);
+    Objects.requireNonNull(domain);
+    Objects.requireNonNull(name);
+    final ExecutionOuterClass.Execution execution =
+        flyteAdminClient.getExecution(project, domain, name);
+    emitFlyteEvents(execution, runState);
+  }
+
+  private void emitFlyteEvents(ExecutionOuterClass.Execution execution, RunState runState) {
+    final Execution.WorkflowExecution.Phase phase = execution.getClosure().getPhase();
+    final FlytePhase flytePhase = FlytePhase.fromProto(phase);
+    try {
+      switch (flytePhase) {
+        case SUCCEEDED:
+          stateManager.receive(Event.terminate(runState.workflowInstance(), Optional.of(0)));
```

Comment:
Can you put that 0 and the 20, 50 and 1 as constant with descriptive names? maybe we have those constants defined already somewhere

Reply:
There was something in RunState which I extended to have the rest of the exit codes as well